GLOSSARY.md

```diff
@@ -1,6 +1,50 @@
 # Glossary
 
-* **Domain Specific Language** (DSL)
-  * The functions and classes that are used by researchers to define a dataset.
-* **Query Model** (QM)
-  * The objects that are created by the DSL.
+**Backend.**
+The logical representation of a collection of EHR data.
+A Backend is associated with a provider of EHR data, a data store, and a Query Language.
+For example, the TPP Backend is associated with TPP, Microsoft SQL Server, and Transact-SQL.
+
+**Contract.**
+A specification for a collection of EHR data.
+A Contract is fulfilled by a Table.
```

Comment:
We don't _really_ have contracts as entity any more. Obviously there is a `contracts/` directory, but all it's got in it are tables just like those in `tables/`.

I think the idea is that a contract would be something like "a table definition which is implemented by more than one EHR provider". But I don't we've fully settled what we're doing here yet.